src/test/java/api/loans/OverrideRenewByBarcodeTests.java

```diff
@@ -0,0 +1,824 @@
+import static api.support.matchers.TextDateTimeMatcher.isEquivalentTo;
+import static api.support.matchers.TextDateTimeMatcher.withinSecondsAfter;
+import static api.support.matchers.ValidationErrorMatchers.hasErrorWith;
+import static api.support.matchers.ValidationErrorMatchers.hasMessage;
+import static api.support.matchers.ValidationErrorMatchers.hasParameter;
+import static java.net.HttpURLConnection.HTTP_OK;
+import static org.hamcrest.CoreMatchers.allOf;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class OverrideRenewByBarcodeTests extends APITests {
+
+  private static final String OVERRIDE_COMMENT = "Comment to override";
+
+  @Test
+  public void canOverrideRenewalRollingLoanFromSystemDate()
+    throws InterruptedException,
+    MalformedURLException,
+    TimeoutException,
+    ExecutionException {
+
+    IndividualResource smallAngryPlanet = itemsFixture.basedUponSmallAngryPlanet();
+    final IndividualResource jessica = usersFixture.jessica();
+
+    final UUID loanId = loansFixture.checkOutByBarcode(smallAngryPlanet, jessica,
+      new DateTime(2018, DateTimeConstants.APRIL, 21, 11, 21, 43))
+      .getId();
+
+    //TODO: Renewal based upon system date,
+    // needs to be approximated, at least until we introduce a calendar and clock
+    DateTime approximateRenewalDate = DateTime.now();
+
+    final JsonObject renewedLoan = loansFixture.overrideRenewalByBarcode(smallAngryPlanet, jessica, OVERRIDE_COMMENT, null).getJson();
+
+    assertThat(renewedLoan.getString("id"), is(loanId.toString()));
+
+    assertThat("user ID should match barcode",
+      renewedLoan.getString("userId"), is(jessica.getId().toString()));
+
+    assertThat("item ID should match barcode",
+      renewedLoan.getString("itemId"), is(smallAngryPlanet.getId().toString()));
+
+    assertThat("status should be open",
+      renewedLoan.getJsonObject("status").getString("name"), is("Open"));
+
+    assertThat("action should be renewed",
+      renewedLoan.getString("action"), is("Renewed through override"));
+
+    assertThat("'actionComment' field should contain comment specified for override",
+      renewedLoan.getString("actionComment"), is(OVERRIDE_COMMENT));
+
+    assertThat("renewal count should be incremented",
+      renewedLoan.getInteger("renewalCount"), is(1));
+
+    assertThat("last loan policy should be stored",
+      renewedLoan.getString("loanPolicyId"),
+      is(APITestSuite.canCirculateRollingLoanPolicyId().toString()));
+
+    assertThat("due date should be approximately 3 weeks after renewal date, based upon loan policy",
+      renewedLoan.getString("dueDate"),
+      withinSecondsAfter(Seconds.seconds(10), approximateRenewalDate.plusWeeks(3)));
+
+    smallAngryPlanet = itemsClient.get(smallAngryPlanet);
+
+    assertThat(smallAngryPlanet, hasItemStatus(CHECKED_OUT));
+  }
+
+  @Test
+  public void canOverrideRenewalRollingLoanFromCurrentDueDate()
```

Comment:
Similar to above, my understanding is that this uses the same behaviour as a regular renewal. 

What is overridden in this scenario (as no due date is provided)?

Reply:
same as above